### ISSUE TITLE
Add Purchase Intent segments for JP test campaign

### ISSUE THREAD

**Issue by moritzhaller** [brave/brave-browser#10235]
Add Purchase Intent segments for JP test campaign

**Comment by LaurenWags** [comment]
Adding `QA/Blocked` label as this is pending uplift merge conflicsts.

**Comment by LaurenWags** [comment]
Uplift completed, should be available with 1.11.96, removing `QA/Blocked`

**Comment by btlechowski** [comment]
`QA/Blocked` on https://github.com/brave/brave-browser/issues/10737

**Comment by btlechowski** [comment]
https://github.com/brave/brave-browser/issues/10737 was merged and is included in 1.11.97.

**Comment by LaurenWags** [comment]
Verified passed with

```
Brave	1.11.97 Chromium: 84.0.4147.89 (Official Build) （64 ビット）
変更履歴	19abfe7bcba9318a0b2a6bc6634a67fc834aa592-refs/branch-heads/4147@{#852}
OS	macOS バージョン10.14.6（ビルド18G3020）
```

* Verified test plan provided to QA by @jsecretan, using Purchase Intent Taxonomy spreadsheet and staging env
* Additional testing done with https://github.com/brave/brave-browser/issues/10737#issuecomment-658730550

<details>
<summary>Confirmed the below scenarios using Japan/Japanese locale configuration</summary>

**Keyword check 1:**
1. Searched on search.yahoo.com for a keyword from the list
2. Saw the below in the logs as expected:
```
[18266:775:0715/090300.116892:VERBOSE1:ads_impl.cc(697)] Extracting purchase intent signal from visited URL
```

3. Saw the below in my `client.json` file:
```
  "purchaseIntentSignalHistory": {
    "jp trial 2-search related word": [
      {
        "timestamp_in_seconds": 1594818180,
        "weight": 1
      }
    ]
  },
```

4. Triggered an ad and saw the below in the logs (note, ad triggers with weight of 1 as expected):
```
[18266:775:0715/090454.458949:VERBOSE1:ads_impl.cc(826)] Serving ad from categories:
[18266:775:0715/090454.459010:VERBOSE1:ads_impl.cc(828)]   jp trial 2-search related word
[18266:775:0715/090454.459834:VERBOSE2:ads_impl.cc(988)] 1 eligible ads with a priority of 1
[18266:775:0715/090454.459887:VERBOSE1:ads_impl.cc(940)] Found 1 eligible ads
[18266:775:0715/090454.459927:VERBOSE3:client.cc(580)] Saving client state
[18266:775:0715/090454.459999:VERBOSE3:client.cc(580)] Saving client state
[18266:775:0715/090454.460100:VERBOSE3:client.cc(580)] Saving client state
[18266:775:0715/090454.460197:VERBOSE3:client.cc(580)] Saving client state
[18266:775:0715/090454.460285:VERBOSE1:ads_impl.cc(1256)] Ad notification shown:
  uuid: 9fda67b9-8756-47e9-b6bb-87887382df36
  parentUuid: 473c63d7-3485-441c-94a5-b230d610750b
  creativeInstanceId: 6a302399-5872-4b29-a245-21920b449a73
  creativeSetId: 5762d2a3-a86e-40fd-a648-1505a8a90e3c
  category: jp trial 2-search related word
```

5. Since the category of `jp trial 2-search related word` matches the the segment of the keyword I searched for, the test passes.

**Keyword check 2:**
1. Searched on google.co.jp for a keyword from the list
2. Saw the below in the logs as expected:
```
[18311:775:0715/091351.619392:VERBOSE1:ads_impl.cc(697)] Extracting purchase intent signal from visited URL
```

3. Saw the below in my `client.json` file:
```
  "purchaseIntentSignalHistory": {
    "jp trial 4-search related word": [
      {
        "timestamp_in_seconds": 1594818831,
        "weight": 1
      }
    ]
  },
```

4. Triggered an ad and saw the below in the logs (note, ad triggers with weight of 1 as expected):
```
[18311:775:0715/091741.110317:VERBOSE1:ads_impl.cc(826)] Serving ad from categories:
[18311:775:0715/091741.110384:VERBOSE1:ads_impl.cc(828)]   jp trial 4-search related word
[18311:775:0715/091741.111393:VERBOSE2:ads_impl.cc(988)] 1 eligible ads with a priority of 1
[18311:775:0715/091741.111450:VERBOSE1:ads_impl.cc(940)] Found 1 eligible ads
[18311:775:0715/091741.111489:VERBOSE3:client.cc(580)] Saving client state
[18311:775:0715/091741.111558:VERBOSE3:client.cc(580)] Saving client state
[18311:775:0715/091741.111616:VERBOSE3:client.cc(580)] Saving client state
[18311:775:0715/091741.111679:VERBOSE3:client.cc(580)] Saving client state
[18311:775:0715/091741.111738:VERBOSE1:ads_impl.cc(1256)] Ad notification shown:
  uuid: cfee994b-cbde-420d-8910-e0623c715b77
  parentUuid: d2068695-0293-471e-829a-c4a462bfe062
  creativeInstanceId: 4f0de61e-ae0f-49d0-8e9b-29b2b7485799
  creativeSetId: 3a9c496e-acae-41dc-9bdd-6dfe4b0bb947
  category: jp trial 4-search related word
```

5. Since the category of `jp trial 4-search related word` matches the the segment of the keyword I searched for, the test passes.

**Website check 1:**
1. Visited an a site from the list
2. Saw the below in my logs as expected:
```
[18375:775:0715/093143.746295:VERBOSE1:ads_impl.cc(697)] Extracting purchase intent signal from visited URL
```

3. Saw the below in my `client.json` file:
```
    "jp trial 1-related content of auto": [
      {
        "timestamp_in_seconds": 1594819903,
        "weight": 1
      }
    ]
  },
```

4. Triggered an ad and saw the below in the logs (note, ad triggers with weight of 1 as expected):
```
[18375:775:0715/093304.620774:VERBOSE1:ads_impl.cc(826)] Serving ad from categories:
[18375:775:0715/093304.620845:VERBOSE1:ads_impl.cc(828)]   jp trial 1-related content of auto
[18375:775:0715/093304.622027:VERBOSE2:ads_impl.cc(988)] 1 eligible ads with a priority of 1
[18375:775:0715/093304.622103:VERBOSE1:ads_impl.cc(940)] Found 1 eligible ads
[18375:775:0715/093304.622173:VERBOSE3:client.cc(580)] Saving client state
[18375:775:0715/093304.622641:VERBOSE3:client.cc(580)] Saving client state
[18375:775:0715/093304.622776:VERBOSE3:client.cc(580)] Saving client state
[18375:775:0715/093304.622840:VERBOSE3:client.cc(580)] Saving client state
[18375:775:0715/093304.622894:VERBOSE1:ads_impl.cc(1256)] Ad notification shown:
  uuid: 0b932015-4c5a-4bfb-9fa9-b4635e55444c
  parentUuid: 03c7266d-46e9-41b1-a3b8-33c09387119a
  creativeInstanceId: 2cdbca84-3c4a-42cc-a8af-9423808517ef
  creativeSetId: a0b2e3e8-8847-4267-adfb-f41e701644ce
  category: jp trial 1-related content of auto
```

5. Since the category of `jp trial 1-related content of auto` matches the the segment of the website I visited, the test passes.

**Website check 2:**
1. Visited an a site from the list
2. Saw the below in my logs as expected:
```
[18424:775:0715/093524.198266:VERBOSE1:ads_impl.cc(697)] Extracting purchase intent signal from visited URL
```

3. Saw the below in my `client.json` file:
```
  "purchaseIntentSignalHistory": {
    "jp trial 4-domain of competitors": [
      {
        "timestamp_in_seconds": 1594820124,
        "weight": 1
      }
    ]
  },
```

4. Triggered an ad and saw the below in the logs (note, ad triggers with weight of 1 as expected):
```
[18424:775:0715/093638.209681:VERBOSE1:ads_impl.cc(826)] Serving ad from categories:
[18424:775:0715/093638.209749:VERBOSE1:ads_impl.cc(828)]   jp trial 4-domain of competitors
[18424:775:0715/093638.210716:VERBOSE2:ads_impl.cc(988)] 1 eligible ads with a priority of 1
[18424:775:0715/093638.210794:VERBOSE1:ads_impl.cc(940)] Found 1 eligible ads
[18424:775:0715/093638.210836:VERBOSE3:client.cc(580)] Saving client state
[18424:775:0715/093638.210901:VERBOSE3:client.cc(580)] Saving client state
[18424:775:0715/093638.210959:VERBOSE3:client.cc(580)] Saving client state
[18424:775:0715/093638.211014:VERBOSE3:client.cc(580)] Saving client state
[18424:775:0715/093638.211069:VERBOSE1:ads_impl.cc(1256)] Ad notification shown:
  uuid: 99ddfde2-d4d9-423f-90d9-2cfb6145c707
  parentUuid: 3f16c342-f9b0-4eb8-820b-ea7f0651e0f4
  creativeInstanceId: 4f0de61e-ae0f-49d0-8e9b-29b2b7485799
  creativeSetId: 3a9c496e-acae-41dc-9bdd-6dfe4b0bb947
  category: jp trial 4-domain of competitors
```

5. Since the category of `jp trial 4-domain of competitors` matches the the segment of the website I visited, the test passes.

</details>

<details>
<summary>Performed a quick regression test of previous segments using US locale configuration</summary>

* Used staging catalog
* Note - segments used to only trigger ads with a weight of 10 or more, this is now 1.

**Keyword check:**
1. Searched on google.com for keyword (`Audi A5`)
2. Saw the below in the logs as expected:
```
[18497:775:0715/094314.981556:VERBOSE1:ads_impl.cc(697)] Extracting purchase intent signal from visited URL
```

3. Saw the below in my `client.json` file:
```
  "purchaseIntentSignalHistory": {
    "automotive purchase intent by category-entry luxury car": [
      {
        "timestamp_in_seconds": 1594820594,
        "weight": 1
      }
    ],
    "automotive purchase intent by make-audi": [
      {
        "timestamp_in_seconds": 1594820594,
        "weight": 1
      }
    ]
  },
```

4. Triggered an ad and saw the below in the logs:
```
[18497:775:0715/094453.163421:VERBOSE1:ads_impl.cc(826)] Serving ad from categories:
[18497:775:0715/094453.163493:VERBOSE1:ads_impl.cc(828)]   automotive purchase intent by make-audi
[18497:775:0715/094453.163537:VERBOSE1:ads_impl.cc(828)]   automotive purchase intent by category-entry luxury car
[18497:775:0715/094453.164716:VERBOSE2:ads_impl.cc(988)] 1 eligible ads with a priority of 1
[18497:775:0715/094453.164777:VERBOSE1:ads_impl.cc(940)] Found 1 eligible ads
[18497:775:0715/094453.164821:VERBOSE3:client.cc(580)] Saving client state
[18497:775:0715/094453.164890:VERBOSE3:client.cc(580)] Saving client state
[18497:775:0715/094453.164950:VERBOSE3:client.cc(580)] Saving client state
[18497:775:0715/094453.165005:VERBOSE3:client.cc(580)] Saving client state
[18497:775:0715/094453.165066:VERBOSE1:ads_impl.cc(1256)] Ad notification shown:
  uuid: 6acc38e2-34bb-4140-9d15-5e4916f6b7a0
  parentUuid: a482b962-8169-46b7-8e78-d69a2406624b
  creativeInstanceId: d9dc0028-8f86-4c92-b25d-981bc1bdb903
  creativeSetId: dea8df38-27b6-4b53-a699-77629934db80
  category: automotive purchase intent by make-audi
```

5. Since the category of `automotive purchase intent by make-audi` matches the the segment of the keyword I searched for, the test passes.

**Website check**
1. Visited an a site (`buyatoyota.com`)
2. Saw the below in my logs as expected:
```
[18565:775:0715/094617.335022:VERBOSE1:ads_impl.cc(697)] Extracting purchase intent signal from visited URL
```

3. Saw multiple entries in `client.json` for `purchaseIntentSignalHistory` as expected.

4. Triggered an ad and saw the below in the logs:
```
[18565:775:0715/094803.449744:VERBOSE1:ads_impl.cc(826)] Serving ad from categories:
[18565:775:0715/094803.449818:VERBOSE1:ads_impl.cc(828)]   automotive-automotive
[18565:775:0715/094803.449861:VERBOSE1:ads_impl.cc(828)]   automotive-pickup trucks
[18565:775:0715/094803.449899:VERBOSE1:ads_impl.cc(828)]   personal finance-personal finance
[18565:775:0715/094803.449937:VERBOSE1:ads_impl.cc(828)]   automotive purchase intent by make-volvo
[18565:775:0715/094803.449974:VERBOSE1:ads_impl.cc(828)]   automotive purchase intent by make-volkswagen
[18565:775:0715/094803.450012:VERBOSE1:ads_impl.cc(828)]   automotive purchase intent by make-toyota
[18565:775:0715/094803.451020:VERBOSE1:ads_impl.cc(846)] No eligible ads found in categories:
[18565:775:0715/094803.451093:VERBOSE1:ads_impl.cc(848)]   automotive-automotive
[18565:775:0715/094803.451136:VERBOSE1:ads_impl.cc(848)]   automotive-pickup trucks
[18565:775:0715/094803.451173:VERBOSE1:ads_impl.cc(848)]   personal finance-personal finance
[18565:775:0715/094803.451209:VERBOSE1:ads_impl.cc(848)]   automotive purchase intent by make-volvo
[18565:775:0715/094803.451244:VERBOSE1:ads_impl.cc(848)]   automotive purchase intent by make-volkswagen
[18565:775:0715/094803.451281:VERBOSE1:ads_impl.cc(848)]   automotive purchase intent by make-toyota
[18565:775:0715/094803.451317:VERBOSE1:ads_impl.cc(879)] Serving ad from parent categories:
[18565:775:0715/094803.451352:VERBOSE1:ads_impl.cc(881)]   automotive
[18565:775:0715/094803.451387:VERBOSE1:ads_impl.cc(881)]   personal finance
[18565:775:0715/094803.451420:VERBOSE1:ads_impl.cc(881)]   automotive purchase intent by make
[18565:775:0715/094803.453488:VERBOSE2:ads_impl.cc(988)] 25 eligible ads with a priority of 1
[18565:775:0715/094803.453596:VERBOSE1:ads_impl.cc(940)] Found 25 eligible ads
[18565:775:0715/094803.453668:VERBOSE3:client.cc(580)] Saving client state
[18565:775:0715/094803.453790:VERBOSE3:client.cc(580)] Saving client state
[18565:775:0715/094803.454270:VERBOSE3:client.cc(580)] Saving client state
[18565:775:0715/094803.454395:VERBOSE3:client.cc(580)] Saving client state
[18565:775:0715/094803.454457:VERBOSE1:ads_impl.cc(1256)] Ad notification shown:
  uuid: ec8986d9-d796-4b92-9812-6ed362df6584
  parentUuid: b6ff4f6b-820f-46d9-b996-bec8272e9432
  creativeInstanceId: bfe0149e-7aa8-4a55-86bc-294e4913f24a
  creativeSetId: 37f7ba07-355d-4d0f-8cdb-b9241550c0c8
  category: personal finance
```

5. No ads configured for any of the areas for this purchase intent classifier, so a different ad was selected. This is expected.

</details>
